openwisp_radius/api/views.py

```diff
@@ -418,15 +448,23 @@ def get(self, request, *args, **kwargs):
 
 
 class RadiusTokenMixin(object):
-    def get_or_create_radius_token(self, user):
-        radius_token, _ = RadiusToken.objects.get_or_create(user=user)
+    def get_or_create_radius_token(self, user, organization=None, expired=False):
+        try:
+            radius_token, _ = RadiusToken.objects.get_or_create(
+                user=user, organization=organization, is_expired=expired
+            )
+        except IntegrityError:
+            radius_token = RadiusToken.objects.get(user=user)
+            radius_token.organization = organization
+            radius_token.is_expired = expired  # TODO:N condition required before chaning value because of validation?
```

Comment:
I want to understand which operations should set `is_expired` as `True` and if by default it should be `True` or `False`.
Also, since this flag is used to check if the token should be used for authentication,
what do you think about renaming `is_auth_allowed` instead?

Reply:
when the radius session expires, the captive portal or radius itself will send an accounting stop request.
We could intercept that and emit a signal, then connect a receiver to that signal to do this.

A good name for this field could be `can_auth` or `is_auth_valid`.
An alternative could be to delete the object, but I'm not 100% sure freeradius does not perform any other operation after the user logs out or the session expires.